### PR TITLE
Add -d (duration) option to argdist, funclatency and syscount

### PR DESCRIPTION
**Pull request by akilesh88**
Adding "-d" (duration) trace option to argdist, funclatency and syscount functionalities. This could be helpful when tracing hot-paths in kernel for specific duration.

### CODE REVIEW

**Comment by yonghong-song** [comment]
[buildbot, ok to test]

**Comment by yonghong-song** [comment]
Thanks for adding this. `-d` definitely useful for these tools. Could you add `-d` to the manpage as well?

**Comment by goldshtn** [comment]
... and the _example.txt files, please :)

File: tools/argdist.py [hunk 1]
```diff
@@ -611,6 +611,8 @@ def __init__(self):
                   help="maximum string size to read from char* arguments")
                 parser.add_argument("-i", "--interval", default=1, type=int,
                   help="output interval, in seconds")
+                parser.add_argument("-d", "--duration",
+                  help="total duration of trace, seconds")
```
Comment (javierhonduco) [inline tools/argdist.py]: What do you think on adding here the type, so we don't have to cast it below? (also applies to the other files :) )

File: tools/argdist.py [hunk 2]
```diff
@@ -684,9 +686,11 @@ def _attach(self):
 
         def _main_loop(self):
                 count_so_far = 0
+                seconds = 0
                 while True:
                         try:
-                                sleep(self.args.interval)
+                                sleep(int(self.args.interval))
```
Comment (javierhonduco) [inline tools/argdist.py]: I think this `int` casting might not be necessary (same for the ones below). argparse should handle that for us already here: https://github.com/iovisor/bcc/pull/1768/files#diff-080a37c8911a2d34cbff11a2f1ca1255R612 😄

File: tools/argdist.py (hunk 1 above)
Comment (akilesh88) [reply tools/argdist.py]: Thanks. I have addressed it in new commit.

**Pull request merged**